src/Console/DuskCommand.php

```diff
@@ -169,15 +192,37 @@ protected function withDuskEnvironment($callback)
 
         $this->writeConfiguration();
 
-        return tap($callback(), function () {
-            $this->removeConfiguration();
+        $this->setupSignalHandler();
+    }
 
-            if (file_exists(base_path($this->duskFile())) && file_exists(base_path('.env.backup'))) {
-                $this->restoreEnvironment();
-            }
+    /**
+     * Setup the SIGINT signal handler for CTRL+C exits.
+     *
+     * @return void
+     */
+    protected function setupSignalHandler()
+    {
+        declare(ticks=1);
+
+        pcntl_signal(SIGINT, function () {
```

Comment:
`ext-pcntl` will need to be required in the `composer.json` file.

Reply:
Added. Perhaps better to add it to a `suggest` like the illuminate/queue package does? What do you think is better in this case? See https://github.com/illuminate/queue/blob/master/composer.json#L41

Would prefer it as a suggestion, but am not sure what a good suggestion text would be.

Reply:
It can't be a suggest because it's always executed when running `php artisan dusk`.

Reply:
True, but if we moved it to suggest if would be together with a `extension_loaded`/`function_exists` call to ensure the extension is loaded there.

But think as Dusk runs only on local machines, the requirement for this extension is minimal.